File: longestsubstring.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res = 1;
        int l = 0, r = 1;
        if (s.length()<=1){
            return s.length();
        }

        while(r<s.length()){
            if(s[r]!=s[l]){
                res += 1;                
            }
            else{
                
            }
            r += 1;
        }
    }
};

File: longestsubstring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) <= 1:
            return len(s)
        maxLen = 0
        l = 0
        charSet = set()
        for r in range(len(s)):
            if s[r] not in charSet:
                charSet.add(s[r])
                maxLen = max(maxLen,r-l+1)
            else:
                while s[r] in charSet:
                    charSet.remove(s[l])
                    l += 1
                charSet.add(s[r])

        return maxLen
        